lib/cublas/linalg.jl

```diff
@@ -24,6 +24,49 @@ function LinearAlgebra.dot(x::StridedCuArray{T}, y::StridedCuArray{T}) where T<:
     dotc(n, x, y)
 end
 
+function LinearAlgebra.dot(x::StridedCuArray{T1}, y::StridedCuArray{T2}) where {T1,T2}
+    n = length(x)
+    n==length(y) || throw(DimensionMismatch("dot product arguments have lengths $(length(x)) and $(length(y))"))
+
+    T = promote_type(T1, T2)
+    res = CuArray(T[zero(T)])
+    MAX_THREADS = 256
+
+    function kernel(x, y, res, T)
+        index = threadIdx().x
+        thread_stride = blockDim().x
+        block_stride = (length(x)-1) ÷ gridDim().x + 1
```

Comment:
You can use `1i32` in CUDA.jl for Int32-literals that often save a couple of registers.

Reply:
Updated